Move all login and account creation page labels to be above inputs

rafh opened the pull request:
There are a few inconsistencies within Gitea and this PR addresses one of them. This PR updates the sign-in page layout, including the register and openID tabs, to match the layout of the settings pages (/user/settings) for more consistency.

This PR updates the following routes:
`/user/login`
`/user/sign_up`
`/user/login/openid`
`/user/forgot_password`
`/user/link_account`
`/user/recover_account`

**Before**
<img width="968" alt="Screenshot 2024-02-05 at 8 27 24 AM" src="https://github.com/go-gitea/gitea/assets/6152817/fb0cb517-57c0-4eed-be1d-56f36bd1960d">


**After**
<img width="968" alt="Screenshot 2024-02-05 at 8 26 39 AM" src="https://github.com/go-gitea/gitea/assets/6152817/428d691d-0a42-4a67-a646-05527f2a7b41">


This PR addresses a revert of the original PR due to this [comment](https://github.com/go-gitea/gitea/pull/28753#issuecomment-1956596817).

KN4CK3R commented:
Could you have a look at the two factor page too? Last time I checked the content was not centered.
![grafik](https://github.com/go-gitea/gitea/assets/1666336/65e39202-f093-4c09-8957-1cf080c58565)

silverwind commented:
Looks cleaner with that layout, I like it.

zokkis commented:
Maybe remove the 2. title?
Because now there is the tab with "sign in" and directly under the tab is the title with "sign in"

wxiaoguang commented on templates/user/auth/captcha.tmpl:
```diff
@@ -1,11 +1,10 @@
 	<div class="required inline field {{if .Err_Captcha}}error{{end}}">
 		<label for="captcha">{{ctx.Locale.Tr "captcha"}}</label>
-		<input id="captcha" name="captcha" value="{{.captcha}}" autocomplete="off">
+		<input id="captcha" class="gt-w-full" name="captcha" value="{{.captcha}}" autocomplete="off">
```
I just realized that it is playing a CSS style overriding game ....

For example: by Fomantic UI: the `inline field` above means that the label is the same row as the input

https://fomantic-ui.com/collections/form.html#inline-field


So, by default, without the `inline`, it should be the UI you need. 


I do not think it is right to use "inline" outside, and change the "inline" behavior inside.

There are already many UI bugs caused by such unclear overridings.

rafh replied:
I understand this perspective. I removed the inline class to allow the default behavior of the `div` to take over. 

On the topic of helper classes, we recently implemented tailwinds. We should update our contribution docs to reflect the usage of these classes? If tailwinds is the path forward, I'll open a PR to update these docs.

https://github.com/go-gitea/gitea/blob/4b8293aa094e725b372329a19da687a6d1550069/docs/content/contributing/guidelines-frontend.en-us.md?plain=1#L50

wxiaoguang replied:
I think it should be updated accordingly. At the moment the plan is to remove most Gitea's own helpers, use Tailwind as much as possible.

lunny commented:
I personally like the previous style. It's more beautiful to not align every line.

wxiaoguang reviewed (changes requested):
Put a "request change" by https://github.com/go-gitea/gitea/pull/29432#discussion_r1503566005

Personally I think it should clearly define the HTML layout and styles, instead of overriding the styles to a different purpose.

If most people agree to override, feel free to dismiss this request change.

Thank you.

wxiaoguang reviewed (commented):
Thank you very much. Without the "inline" patches, the new approach looks better to me.

* It seems that the ".user.activate" form is still using "inline", should it be updated too?
* Why sometimes there is "md:tw-max-w-2xl tw-m-auto" but sometimes there isn't? Could there be some general (unified) framework approaches?

rafh commented:
> * md:tw-max-w-2xl

This [line of CSS](https://github.com/go-gitea/gitea/pull/29432/files#diff-27543749149b34d4bb8459b8e24c992af84863721d6443a53d95ec0e43f39860L246
) is the reason for the use of `md:tw-max-w-2xl tw-m-auto`. To create a unified approach, I removed the CSS and updated the templates that used it. The PR grew, but things are now more consistent.

wxiaoguang commented:
Thank you, it looks better. Actually, I am not a fan of fine-tuning everything, that's why I proposed to use some clearly defined CSS classes for these forms. But the current approach is not unacceptable ....

The last question: why `user.activate form` is left there? I think it also belongs to the "user sign in/out" pages.

silverwind commented:
Sidenote: Please don't raise PR's from `main` branch for [reasons mentioned here](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/) and because it's a pain for reviewers to work with that branch as they need to rename their local branch.
